File: coding/usaco/namenum/namenum.java
/*
ID: rohitya1
LANG: JAVA
PROG: namenum
*/

import java.util.*;
import java.io.*;
import java.lang.*;

class namenum {

    private static char lookup(char ch) {
        switch(ch) {
            case 'A': case 'B': case 'C': return '2';
            case 'D': case 'E': case 'F': return '3';
            case 'G': case 'H': case 'I': return '4';
            case 'J': case 'K': case 'L': return '5';
            case 'M': case 'N': case 'O': return '6';
            case 'P': case 'R': case 'S': return '7';
            case 'T': case 'U': case 'V': return '8';
            case 'W': case 'X': case 'Y': return '9';
        }
        return '0';
    }

    public static void main(String[] args) throws IOException {
        BufferedReader in = new BufferedReader(new FileReader("namenum.in"));
        BufferedReader dictin = new BufferedReader(new FileReader("dict.txt"));
        PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("namenum.out")));

        String brand = in.readLine();
        int results = 0;
        String word;
        while ((word = dictin.readLine()) != null) {
            String key = "";
            boolean pass = false;
            for(char ch: word.toCharArray()) {
                if (ch == 'Z' || ch == 'Q') {
                    pass = true; break;
                }
                key += namenum.lookup(ch);
            }
            if (!pass && key.equals(brand)) {
                results++;
                out.println(word);
            }
        }
        if (results == 0) out.println("NONE");

        in.close();
        out.close();
        dictin.close();
        System.exit(0);
    }
}

File: coding/usaco/namenum/namenum.cpp
/*
ID: rohitya1
LANG: C++11
PROG: namenum
*/

#include <iostream>
#include <fstream>
#include <string>
#include <map>

using namespace std;

int main() {
    ifstream fin("namenum.in");
    ifstream dictin("dict.txt");
    ofstream fout("namenum.out");

    string brand;
    fin >> brand;

    map<char, char> lookup;
    multimap<string, string> dict;

    lookup['A'] = lookup['B'] = lookup['C'] = '2';
    lookup['D'] = lookup['E'] = lookup['F'] = '3';
    lookup['G'] = lookup['H'] = lookup['I'] = '4';
    lookup['J'] = lookup['K'] = lookup['L'] = '5';
    lookup['M'] = lookup['N'] = lookup['O'] = '6';
    lookup['P'] = lookup['R'] = lookup['S'] = '7';
    lookup['T'] = lookup['U'] = lookup['V'] = '8';
    lookup['W'] = lookup['X'] = lookup['Y'] = '9';

    string word;
    while (!dictin.eof()) {
        dictin >> word;
        bool pass = false;
        string key = "";
        for (int i = 0; i < word.length(); i++) {
            char ch = word.at(i);
            if (ch == 'Z' || ch == 'Q') { pass = true; break;}
            key += lookup[ch];
        }
        if (!pass) {
            dict.insert(pair<string, string>(key, word));
        }
    }
    dictin.close();

    auto range = dict.equal_range(string(brand));
    int result = 0;
    for (auto it = range.first; it != range.second; it++) {
        fout << it->second << endl;
        result++;
    }
    if (result == 0) fout << "NONE" << endl;

    fin.close();
    fout.close();
    return 0;
}
